hbase-mapreduce/src/main/java/org/apache/hadoop/hbase/mapreduce/TableOutputFormat.java

```diff
@@ -91,6 +115,21 @@ public class TableOutputFormat<KEY> extends OutputFormat<KEY, Mutation> implemen
   /** The configuration. */
   private Configuration conf = null;
 
+  private static Connection createConnection(Configuration conf) throws IOException {
+    String outputCluster = conf.get(OUTPUT_CLUSTER);
+    if (!StringUtils.isBlank(outputCluster)) {
+      URI uri;
+      try {
+        uri = new URI(outputCluster);
+      } catch (URISyntaxException e) {
+        throw new IOException("malformed connection uri: " + outputCluster, e);
```

Comment:
nit: add the configuration key to the error message to help the user track down the issue? There's a couple places where this pattern repeats.